### PR TITLE
Refactor Register

### PR DESCRIPTION
**Pull request by Tishasoumya-02**
#4460

### CODE REVIEW

File: src/components/theme/Register/Register.jsx [hunk 1]
```diff
@@ -58,145 +55,83 @@ const messages = defineMessages({
...
+    </div>
+  );
+};
+
+Register.propTypes = {
+  createUser: PropTypes.func.isRequired,
+  loading: PropTypes.bool.isRequired,
+  loaded: PropTypes.bool.isRequired,
+  error: PropTypes.shape({
+    message: PropTypes.string,
```
Comment (nileshgulia1) [inline src/components/theme/Register/Register.jsx:129]: You can remove these propTypes.

File: src/components/theme/Register/Register.jsx [hunk 2]
```diff
@@ -58,145 +55,83 @@ const messages = defineMessages({
...
-  connect(
-    (state) => ({
-      loading: state.users.create.loading,
-      loaded: state.users.create.loaded,
-      error: state.users.create.error,
-    }),
-    { createUser },
-  ),
-)(Register);
+export default compose(withRouter)(Register);
```
Comment (nileshgulia1) [inline src/components/theme/Register/Register.jsx:137]: withRouter will be deprecated in react-router v6 in favour of useLocation , useHistory hooks. I think we should refrain from using it.

File: src/components/theme/Register/Register.test.jsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ describe('Register', () => {
     const component = renderer.create(
       <Provider store={store}>
         <MemoryRouter>
-          <Register />
+          <Register createUser={createUser} loaded={false} loading={false} />
```
Comment (nileshgulia1) [inline src/components/theme/Register/Register.test.jsx:29]: same as in https://github.com/plone/volto/pull/4861/files#r1233820804

**Review by nileshgulia1 — CHANGES_REQUESTED** [review]
Let's fix the comments.

File: src/components/theme/Register/Register.jsx [hunk 4]
```diff
@@ -1,19 +1,13 @@
-/**
- * Register container.
- * @module components/theme/Register/Register
- */
-
-import React, { Component } from 'react';
+import { useEffect, useState } from 'react';
 import { Helmet } from '@plone/volto/helpers';
```
Comment (nileshgulia1) [inline src/components/theme/Register/Register.jsx:2]: Please fix this import.

**Comment by JeffersonBledsoe** [comment]
@Tishasoumya-02 The refactor looks good here, but the actual component seems like it was broken before refactoring.
It is unable to actually succeed in a form submission due to the backend call return an error as the form doesn't send a `password` value or the `sendPasswordReset` option. If you could write a test that should pass (i.e. the form sends successfully and the toast appears) but actually fails that would be great and we can go from there on getting this merged.

cc @sneridagh @nileshgulia1 For visibility

## Returned error

```
{
  "error": {
    "errors": [
      {
        "field": "username", 
        "message": "Property 'User Name' is required."
      }, 
      {
        "field": "sendPasswordReset", 
        "message": "You have to either send a password or sendPasswordReset."
      }
    ], 
    "message": "Error in fields. Property 'User Name' is required. You have to either send a password or sendPasswordReset.", 
    "type": "WrongParameterError"
  }
}
```

**Review by nileshgulia1 — APPROVED** [review]
I approve it on the context of refactor. We still have to take care of original fix, maybe in a different PR?

**Comment by JeffersonBledsoe** [comment]
@plone/volto-team Updated to the lastest main. Happy for this to be merged?

File: packages/volto/news/4861.feature [hunk 5]
```diff
@@ -0,0 +1 @@
+Refactor Register Component -@Tishasoumya-02
```
[the remark this answers is not in the document]
Comment (ichim-david) [reply packages/volto/news/4861.feature:1]: @Tishasoumya-02 you will need for your other pull requests to modify the changelog.

In summary, the changelog needs to have two characteristics:
1. a bit more information as Refactor Component isn't enough to say what was refactored
2. The changelog ends with a period and not a dash.

This is my proposal, @stevepiercy what do you think about my proposal? Anything you would add?

[the remark this answers is not in the document]
Comment (stevepiercy) [reply packages/volto/news/4861.feature:1]: I edited it for grammar and to be concise.

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
I restarted failed checks, which I think are flaky.

I also created a new issue from @JeffersonBledsoe's comment in https://github.com/plone/volto/issues/6233.

This PR is concerned only with refactoring.

**Comment by stevepiercy** [comment]
@Tishasoumya-02 I'm not clear what that error has to do with this PR. Can you elaborate?

Also please [do not paste images of error messages](https://meta.stackoverflow.com/questions/285551/why-should-i-not-upload-images-of-code-data-errors). Paste the entire text into a comment surrounded by triple backticks to format it.

````
```
error message
```
````

**Comment by Tishasoumya-02** [comment]
Sorry for the noise ! I was getting this error every time I was trying to create a new document, after `make clean` then `make install` , it started working !

**Comment by Tishasoumya-02** [comment]
> @Tishasoumya-02 I'm not clear what that error has to do with this PR. Can you elaborate?
> 
> Also please [do not paste images of error messages](https://meta.stackoverflow.com/questions/285551/why-should-i-not-upload-images-of-code-data-errors). Paste the entire text into a comment surrounded by triple backticks to format it.
> 
> ```
> ```
> error message
> ```
> ```

Yes, will keep in mind! Thanks :)

**Review by davisagli — APPROVED** [review]

**Pull request merged**